sdk/storage/azure-storage-files-shares/src/share_responses.cpp

```diff
@@ -60,4 +61,115 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     };
   }
 
+  void ListSharesPagedResponse::OnNextPage(const Azure::Core::Context& context)
+  {
+    auto protocolLayerOptions = _detail::ShareRestClient::Service::ListSharesSinglePageOptions();
```

Comment:
Why is creating a copy of the options required here rather than using the options passed in to the ListX API directly? Add a comment.

This coding pattern introduces some brittleness if we add new options in the future. Can we make a copy ctor instead?

Reply:
Oh, this is creating a `ListSharesSinglePageOptions` which is a different type. Why do we even have the `SinglePage` option things anymore? This makes maintenance harder since it might cause confusion in the future.

Reply:
It's protocol layer code, auto-generated. We can rename it and remove the `SinglePage` suffix. It's in details namespace, invisible to customers, not a breaking change anyway.

Reply:
> We can rename it and remove the `SinglePage` suffix.

How? Wouldn't that type then collide with the existing.

>  It's in details namespace, invisible to customers, not a breaking change anyway.

I get it, that's not my concern. I am trying to understand if we actually need this separate options type, at all. If it just here for now as a stop gap and we plan to remove it, that's fine. How would that work?

Reply:
> How? Wouldn't that type then collide with the existing.

No, it's in different namespace.

> I get it, that's not my concern. I am trying to understand if we actually need this separate options type, at all. If it just here for now as a stop gap and we plan to remove it, that's fine. How would that work?

Because in convenience layer, we have mandatory options in parameter list, while optional ones in the option bag. But being mandatory or optional is a very subjective thing, and it definitely requires human intervention to decide. This is very hard for auto-generator. So for generated protocol layer code, we just have **every** option in the option bag.